csh_auth.go

```diff
@@ -0,0 +1,156 @@
+package csh_auth
```

Comment:
nit: run `go fmt` to properly order imports, and fix alignments on struct definitions.